### PR TITLE
Add a diagnostic dialog for missing variables

### PR DESCRIPTION
**Pull request by D8H**
### Changes
- Show a diagnostic report when a preview is launched and there are missing:
  - scene variables
  - object variables
  - behaviors

### CODE REVIEW

File: Core/GDCore/Events/Parsers/ExpressionParser2.h [hunk 1]
```diff
@@ -547,28 +547,28 @@ class GD_CORE_API ExpressionParser2 {
   }
   ///@}
 
-  std::unique_ptr<ExpressionParserDiagnostic> ValidateOperator(
+  std::unique_ptr<ExpressionParserError> ValidateOperator(
       gd::String::value_type operatorChar) {
     if (operatorChar == '+' || operatorChar == '-' || operatorChar == '/' ||
         operatorChar == '*') {
-      return gd::make_unique<ExpressionParserDiagnostic>();
+      return std::unique_ptr<ExpressionParserError>(nullptr);
```
Comment (D8H) [inline Core/GDCore/Events/Parsers/ExpressionParser2.h:554]: This change is not just a refactoring. It was wrongly replacing the null pointer with a default constructed instance.
When removing the base class with `IsError() { return false }`, it returned `true` which causes bugs.
`IsError()` no longer exist, only null is checked.

File: Core/GDCore/Events/Parsers/ExpressionParser2Node.cpp [hunk 2]
```diff
@@ -6,5 +6,5 @@
 #include "ExpressionParser2Node.h"
 
 namespace gd {
-gd::String ExpressionParserDiagnostic::noMessage = "";
+
 }
```
Comment (D8H) [inline Core/GDCore/Events/Parsers/ExpressionParser2Node.cpp:10]: This file is now empty, but I guess it may be useful one day.
Should I remove it?

Comment (4ian) [reply Core/GDCore/Events/Parsers/ExpressionParser2Node.cpp:10]: You can keep it :)

File: Core/GDCore/IDE/Events/ExpressionValidator.cpp [hunk 3]
```diff
@@ -247,9 +247,10 @@ ExpressionValidator::Type ExpressionValidator::ValidateFunction(
   if (!function.behaviorName.empty() &&
       !objectsContainersList.HasBehaviorInObjectOrGroup(function.objectName,
                                       function.behaviorName)) {
-    RaiseTypeError(_("This behavior is not attached to this object."),
-                   function.behaviorNameLocation,
-                   /*isFatal=*/false);
+    RaiseError(gd::ExpressionParserError::ErrorType::MissingBehavior,
+               _("This behavior is not attached to this object."),
+               function.behaviorNameLocation,
+               /*isFatal=*/false, function.behaviorName, function.objectName);
```
[the remark this answers is not in the document]
Comment (4ian) [reply Core/GDCore/IDE/Events/ExpressionValidator.cpp:253]: Ok!

File: Core/GDCore/Events/Parsers/ExpressionParser2.h (hunk 1 above)
Comment (4ian) [reply Core/GDCore/Events/Parsers/ExpressionParser2.h:554]: Thanks for the fix 👍 Much better

**Pull request merged**